src/main/kotlin/org/wa/rceditor/application/model/MainModel.kt

```diff
@@ -1,26 +1,310 @@
 package org.wa.rceditor.application.model
 
+import javafx.beans.property.SimpleObjectProperty
+import javafx.collections.ObservableList
+import org.wycliffeassociates.resourcecontainer.ResourceContainer
+import org.wycliffeassociates.resourcecontainer.entity.Checking
+import org.wycliffeassociates.resourcecontainer.entity.DublinCore
+import org.wycliffeassociates.resourcecontainer.entity.Manifest
+import tornadofx.*
+import java.io.File
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
 import kotlin.system.exitProcess
 
 class MainModel {
```

Comment:
Joe made a good point the other day about MVVM that might be relevant here. MVVM is about separating the view from the model (domain logic). In Clean Architecture terms, that means separating the view layout and event handlers (which just call the view model) from the domain layer use cases. The ViewModel is the piece that connects the view presentation with the domain layer use cases.

In this instance, that might mean you can move the properties from the model into the ViewModel and move any business logic from the model to one or more domain-layer use cases. Then the ViewModel is responsible for triggering those use cases based on user input or using them to load in the data. If you move in this direction, you could eliminate the MainModel class since it would be replaced by domain-layer use cases that have the business logic. (Business logic could include validating fields, as an example).

Reply:
@mbr4477 Good point!